<center><h1>Técnicas de Inteligencia Artificial</header1></center>

Presentado por: María Párraga  <br>
Fecha: 15/07/2022

# Importación de librerias necesarias

In [7]:
#Para esta actividad se importarán las siguientes librerías:
#importar libreria pandas
import pandas as pd
# importar metodo de scalado standar
from sklearn.preprocessing import StandardScaler
# Importación de método para dividir datos de prueba 
from sklearn.model_selection import train_test_split
# Importación de método para clasificación
from sklearn.tree import DecisionTreeClassifier
#Importación de modelo
from sklearn import linear_model
#Importacion de metodos para calculo de error cuadratico y variaza
from sklearn.metrics import mean_squared_error, r2_score

## Cargar el Dataset

In [8]:
#nombre del Dataset
data = 'Real estate.csv'
#Transfromando datos cvs con delimitador , 
data2 = pd.read_csv(data, header=0, sep=',')

#Creando un nuevo dataset
df = pd.DataFrame(data2)
#imprimir los datos del dataset
df.head()


No  X1 transaction date  X2 house age  \
0   1             2012.917          32.0   
1   2             2012.917          19.5   
2   3             2013.583          13.3   
3   4             2013.500          13.3   
4   5             2012.833           5.0   

   X3 distance to the nearest MRT station  X4 number of convenience stores  \
0                                84.87882                               10   
1                               306.59470                                9   
2                               561.98450                                5   
3                               561.98450                                5   
4                               390.56840                                5   

   X5 latitude  X6 longitude  Y house price of unit area  
0     24.98298     121.54024                        37.9  
1     24.98034     121.53951                        42.2  
2     24.98746     121.54391                        47.3  
3     24.98746     121.54391                        54.8  
4     24.97937     121.54245                        43.1

## Descripción de la fuente del Dataset

Datos extraidos de la pagina web Kagle (https://www.kaggle.com/datasets/quantbruce/real-estate-price-prediction/code?resource=download), los mmismos que son datos reales del precio de una propiedad por unidad de área tomada en diferentes años, localizaciones y con diferentes características. El Dataset es utilizado para desarrollar modelos de regresión lineal multiple y obtener predicciones del precio de una propiedad a partir de sus características.

## Explique el problema a resolver. 
El problema a resolver es la predicción del precio por unidad de área de una propiedad dada característica principales como su localización, año de transacción, cercania a tiendas, estación de transporte. La modelación de una función con multiples variables permite la predicción de un terreno y calcular su plusvalia a lo largo de los años, permitiendo aplicarlo en un mercado de inversión en bienes raíces, pues conocer el precio que tendra en el futuro permite al propietario decidir si vender o no. Otra potencial aplicación es en la hipotéca de la propiedad, pues si se conoce el precio a futuro de un terreno la entidad financiera puede asegurar la garantia del solicitande e inclusive proporcionar un prestamo de mayor valor.

## Caracterización del Dataset

Realice una descripción de los datos con:

>- Número de instancias en total.
>- Número de atributos de entrada, su significado y tipo.
>- Estadísticas de la variable objetivo.
>- Estadísticas los atributos en relación con la variable objetivo.


In [9]:
#Código que responde a la descripción anterior
df.describe()

No  X1 transaction date  X2 house age  \
count  414.000000           414.000000    414.000000   
mean   207.500000          2013.148971     17.712560   
std    119.655756             0.281967     11.392485   
min      1.000000          2012.667000      0.000000   
25%    104.250000          2012.917000      9.025000   
50%    207.500000          2013.167000     16.100000   
75%    310.750000          2013.417000     28.150000   
max    414.000000          2013.583000     43.800000   

       X3 distance to the nearest MRT station  \
count                              414.000000   
mean                              1083.885689   
std                               1262.109595   
min                                 23.382840   
25%                                289.324800   
50%                                492.231300   
75%                               1454.279000   
max                               6488.021000   

       X4 number of convenience stores  X5 latitude  X6 longitude  \
count                       414.000000   414.000000    414.000000   
mean                          4.094203    24.969030    121.533361   
std                           2.945562     0.012410      0.015347   
min                           0.000000    24.932070    121.473530   
25%                           1.000000    24.963000    121.528085   
50%                           4.000000    24.971100    121.538630   
75%                           6.000000    24.977455    121.543305   
max                          10.000000    25.014590    121.566270   

       Y house price of unit area  
count                  414.000000  
mean                    37.980193  
std                     13.606488  
min                      7.600000  
25%                     27.700000  
50%                     38.450000  
75%                     46.600000  
max                    117.500000

Los datos tomados en un analisis general nos reflejan 414 de datos recolectados para el estudio, dentro de los cuales una desviación estandar pequea en la mayoria de las variables, lo que radica que los datos no poseen aparenteente cambios significativos en su recolecciónes, trayendo un aalisis de datos concentrados mas no exparsidos de una tendencia. Se debe tener en cuenta que os dotos son recolecciones de los años 2012 y 2013, que si bien esta variable se la puede considerar como factor, al ser solo 2 años no se puede asegurar confianza si es utilizada como variable principal para una predicción en oños diferentes.

Los datos más relevantes se puede ver que el promedio del precio obtenido por unidad de área es de 37.98 siendo el mayor precio de 117.5, observando una gran diferencia de precio, lo que conlleva al razonamiento que las propiedades que poseen radican fuertemente el precio del mismo.

### Número de atributos de entrada, su significado y tipo.<br>
**X1 transaction date** --> Categórico: Fecha de transacción de realización de la transacción<br>
**X2 house age**--> Continuo: Edad de la casa o propiedad<br>
**X3 distance to the nearest MRT station** --> Continuo: distancia a la estación MRT<br>
**X4 number of convenience stores** -->  Continuo:Número de tiendas cercanas a la propiedad<br>
**X5 latitude** --> Continuo: Latitud de la localización de la propiedad<br>
**X6 longitude** --->Continuo: Longitud de la localización de la propiedad<br>
**Y house price of unit area** --->Continuo: Precio por unidad de área de la propiedad<br>

**Variable objetivo:** Y house price of unit area <br>
**Variables de entrada:** X2 house age, X3 distance to the nearest MRT station, X4 number of convenience stores, X5 latitude, X6 longitude 

## Preprocesamiento del dataset. Transformaciones previas necesarias para la modelación

## División del dataset en datos de entrenamiento y datos de test 

In [2]:
#Código que realice la división en entrenamiento y test, de acuerdo con la estretgia de evaluación planeada. Describa cuál es.


## Regresión Lineal

In [3]:
#Códo del modelo

In [4]:
#Entrenamiento


In [5]:
#Test


## Evaluación del Modelo